diff -ur ffmpeg-src/libavcodec/dxva2.h ffmpeg-src-new/libavcodec/dxva2.h
--- ffmpeg-src/libavcodec/dxva2.h	2011-03-17 17:33:25.660024200 -0500
+++ ffmpeg-src-new/libavcodec/dxva2.h	2023-03-25 00:31:23.461834620 -0500
@@ -24,7 +24,7 @@
 #define AVCODEC_DXVA_H
 
 #include <stdint.h>
-
+#include <d3d9.h>
 #include <dxva2api.h>
 
 /**
diff -ur ffmpeg-src/libavcodec/dxva2_internal.h ffmpeg-src-new/libavcodec/dxva2_internal.h
--- ffmpeg-src/libavcodec/dxva2_internal.h	2011-03-17 17:33:25.660024200 -0500
+++ ffmpeg-src-new/libavcodec/dxva2_internal.h	2023-03-25 00:38:34.669814198 -0500
@@ -26,6 +26,7 @@
 #define _WIN32_WINNT 0x0600
 #define COBJMACROS
 #include "dxva2.h"
+#include <dxva.h>
 #include "avcodec.h"
 #include "mpegvideo.h"
 
diff -ur ffmpeg-src/libavcodec/x86/h264_qpel_mmx.c ffmpeg-src-new/libavcodec/x86/h264_qpel_mmx.c
--- ffmpeg-src/libavcodec/x86/h264_qpel_mmx.c	2011-03-17 17:33:25.847224500 -0500
+++ ffmpeg-src-new/libavcodec/x86/h264_qpel_mmx.c	2023-03-24 23:55:49.293935696 -0500
@@ -398,7 +398,7 @@
             "2:                         \n\t"\
             \
             : "+a"(src), "+c"(dst)\
-            : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "g"(h)\
+            : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "rm"(h)\
             : "memory"\
         );\
         src += 4-(h+5)*srcStride;\
@@ -446,7 +446,7 @@
             QPEL_H264HV(%%mm3, %%mm4, %%mm5, %%mm0, %%mm1, %%mm2, 15*48)\
             "2:                     \n\t"\
             : "+a"(src)\
-            : "c"(tmp), "S"((x86_reg)srcStride), "g"(size)\
+            : "c"(tmp), "S"((x86_reg)srcStride), "rm"(size)\
             : "memory"\
             );\
         tmp += 4;\
@@ -823,7 +823,7 @@
         "2:                          \n\t"\
         \
         : "+a"(src), "+c"(dst)\
-        : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "g"(h)\
+        : "S"((x86_reg)srcStride), "D"((x86_reg)dstStride), "rm"(h)\
         : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3", \
                        "%xmm4", "%xmm5", "%xmm6", "%xmm7",)\
           "memory"\
@@ -878,7 +878,7 @@
             QPEL_H264HV_XMM(%%xmm3, %%xmm4, %%xmm5, %%xmm0, %%xmm1, %%xmm2, 15*48)
             "2:                         \n\t"
             : "+a"(src)
-            : "c"(tmp), "S"((x86_reg)srcStride), "g"(size)
+            : "c"(tmp), "S"((x86_reg)srcStride), "rm"(size)
             : XMM_CLOBBERS("%xmm0", "%xmm1", "%xmm2", "%xmm3",
                            "%xmm4", "%xmm5", "%xmm6", "%xmm7",)
               "memory"
diff -ur ffmpeg-src/libavformat/os_support.h ffmpeg-src-new/libavformat/os_support.h
--- ffmpeg-src/libavformat/os_support.h	2011-03-17 17:33:25.956424700 -0500
+++ ffmpeg-src-new/libavformat/os_support.h	2023-03-25 00:26:46.821847722 -0500
@@ -58,6 +58,8 @@
 #if !HAVE_POLL_H
 typedef unsigned long nfds_t;
 
+#include <winsock2.h>
+#if 0
 struct pollfd {
     int fd;
     short events;  /* events to look for */
@@ -77,7 +79,7 @@
 #define POLLERR    0x0004  /* errors pending */
 #define POLLHUP    0x0080  /* disconnected */
 #define POLLNVAL   0x1000  /* invalid file descriptor */
-
+#endif
 
 int poll(struct pollfd *fds, nfds_t numfds, int timeout);
 #endif /* HAVE_POLL_H */
